### PR TITLE
ocivip: new resource agent

### PR DESCRIPTION
**Pull request by garutilorenzo**
New resource agent: ocivip

This new resource agent allow to manage secondary private ip addresses for Compute instances
in the Oracle Cloud infrastructure (OCI)

### CODE REVIEW

File: heartbeat/ocivip [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+INSTANCE_ID="$(curl -s -H "Authorization: Bearer Oracle" -L http://169.254.169.254/opc/v2/instance | jq -r '.id')"
+VNIC_ID="$(curl -s -H "Authorization: Bearer Oracle" -L http://169.254.169.254/opc/v2/vnics/ | jq -r '.[0].vnicId')"
+COMPARTMENT_ID=$(curl -s -H "Authorization: Bearer Oracle" -L http://169.254.169.254/opc/v2/instance | jq -r '.compartmentId')
+AVAILABILITY_DOMAIN=$(curl -s -H "Authorization: Bearer Oracle" -L http://169.254.169.254/opc/v2/instance | jq -r '.availabilityDomain')
+PRIMARY_IFACE=$(ip -4 route ls | grep default | grep -Po '(?<=dev )(\S+)')
+
+
+case $__OCF_ACTION in
+    start)
+        ocivip_validate
```
Comment (oalbrigt) [inline heartbeat/ocivip:203]: You need `|| exit $?` here, or it wont fail unless exit is used in validate().

File: heartbeat/ocivip [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+# limitations under the License.
+#
+#
+
+#
+#  Prerequisites:
+#
+#  - OCI CLI installed (https://docs.oracle.com/en-us/iaas/Content/API/SDKDocs/climanualinst.htm) 
+#  - dynamic group with a policy attacched
+#  - the policy must have this statement:
```
Comment (oalbrigt) [inline heartbeat/ocivip:28]: You should probably mention this in the longdesc section of the metadata.

File: heartbeat/ocivip [hunk 3]
```diff
@@ -0,0 +1,238 @@
...
+    RET=$?
+
+    if [ $RET -ne 0 ]; then
+        return $OCF_NOT_RUNNING
+    fi
+    return $OCF_SUCCESS
+}
+
+ocivip_validate() {
+    check_binary ${OCICLI}
```
Comment (oalbrigt) [inline heartbeat/ocivip:170]: You should also check for `jq` here.

File: heartbeat/ocivip [hunk 4]
```diff
@@ -0,0 +1,238 @@
...
+        ;;
+esac
+
+OCICLI="${OCF_RESKEY_ocicli}"
+SECONDARY_PRIVATE_IP="${OCF_RESKEY_secondary_private_ip}"
+INSTANCE_ID="$(curl -s -H "Authorization: Bearer Oracle" -L http://169.254.169.254/opc/v2/instance | jq -r '.id')"
+VNIC_ID="$(curl -s -H "Authorization: Bearer Oracle" -L http://169.254.169.254/opc/v2/vnics/ | jq -r '.[0].vnicId')"
+COMPARTMENT_ID=$(curl -s -H "Authorization: Bearer Oracle" -L http://169.254.169.254/opc/v2/instance | jq -r '.compartmentId')
+AVAILABILITY_DOMAIN=$(curl -s -H "Authorization: Bearer Oracle" -L http://169.254.169.254/opc/v2/instance | jq -r '.availabilityDomain')
+PRIMARY_IFACE=$(ip -4 route ls | grep default | grep -Po '(?<=dev )(\S+)')
```
Comment (oalbrigt) [inline heartbeat/ocivip:198]: It seems like you arent using all of these.

**Review by oalbrigt — CHANGES_REQUESTED** [review]
Looks good in general.

I left some notes for improvements.

File: heartbeat/ocivip (hunk 1 above)
Comment (garutilorenzo) [reply heartbeat/ocivip]: Fixed [here](https://github.com/ClusterLabs/resource-agents/pull/1751/commits/3d6085f0789b3ea14a1a10f23ed00fdcd4d0c95e)

File: heartbeat/ocivip (hunk 3 above)
Comment (garutilorenzo) [reply heartbeat/ocivip:201]: Fixed [here](https://github.com/ClusterLabs/resource-agents/pull/1751/commits/a76312ed28a0f9ce3ed19a7bf111443d5ba2f5e1)

File: heartbeat/ocivip (hunk 4 above)
Comment (garutilorenzo) [reply heartbeat/ocivip:224]: Fixed [here](https://github.com/ClusterLabs/resource-agents/pull/1751/commits/5a5fefb1e8c5084c75335336cce4e5a7c43bec8a)

File: heartbeat/ocivip (hunk 2 above)
Comment (garutilorenzo) [reply heartbeat/ocivip:29]: Fixed [here](https://github.com/ClusterLabs/resource-agents/pull/1751/commits/a76312ed28a0f9ce3ed19a7bf111443d5ba2f5e1)

**Comment by oalbrigt** [comment]
ok to test

**Review by oalbrigt — APPROVED** [review]

**Comment by oalbrigt** [comment]
Thanks.

**Pull request merged**